[23.05] mediatek: filogic: add support for ASUS RT-AX59U

xavifr opened the pull request:
(based on support for ASUS RT-AX59U by liushiyou006)

SOC: MediaTek MT7986
RAM: 512MB DDR4
FLASH: 128MB SPI-NAND (Winbond W25N01GV)
WIFI: Mediatek MT7986 DBDC 802.11ax 2.4/5 GHz
ETH: MediaTek MT7531 Switch
UART: 3V3 115200 8N1 (Pinout silkscreened / Do not connect VCC)

Upgrade from AsusWRT to OpenWRT using UART

    Download the OpenWrt initramfs image.
    Copy the image to a TFTP server reachable at 192.168.1.70/24. Rename the image to rtax59u.bin.

    Connect the PC with TFTP server to the RT-AX59U.
    Set a static ip on the ethernet interface of your PC.
    (ip address: 192.168.1.70, subnet mask:255.255.255.0)
    Conect to the serial console, interrupt the autoboot process by pressing '4' when prompted.

    Download & Boot the OpenWrt initramfs image.

    $ setenv ipaddr 192.168.1.1
    $ setenv serverip 192.168.1.70
    $ tftpboot 0x46000000 rtax59u.bin
    $ bootm 0x46000000

    Wait for OpenWrt to boot. Transfer the sysupgrade image to the device using scp and install using sysupgrade.

    $ sysupgrade -n <path-to-sysupgrade.bin>

Upgrade from AsusWRT to OpenWRT using WebUI

    Download transit TRX file from https://drive.google.com/drive/folders/1A20QdjK7Udagu31FSszpWAk8-cGlCwsq

    Upgrade firmware from WebUI (192.168.50.1) using downloaded TRX file

    Wait for OpenWRT to boot (192.168.1.1).

    Upgrade system with sysupgrade image using luci or uploading it through scp and executing sysupgrade command

MAC Address for WLAN 5g is not following the same algorithm as in AsusWRT. We have increased by one the WLAN 5g to avoid collisions with other networks from WLAN 2g when bit 28 is already set.

                : Stock             : OpenWrt
    WLAN 2g (1) : C8:xx:xx:0D:xx:D4 : C8:xx:xx:0D:xx:D4
    WLAN 2g (2) :                   : CA:xx:xx:0D:xx:D4
    WLAN 2g (3) :                   : CE:xx:xx:0D:xx:D4
    WLAN 5g (1) : CA:xx:xx:1D:xx:D4 : CA:xx:xx:1D:xx:D5
    WLAN 5g (2) :                   : CE:xx:xx:1D:xx:D5
    WLAN 5g (3) :                   : C2:xx:xx:1D:xx:D5
    
    WLAN 2g (1) : 08:xx:xx:76:xx:BE : 08:xx:xx:76:xx:BE
    WLAN 2g (2) :                   : 0A:xx:xx:76:xx:BE
    WLAN 2g (3) :                   : 0E:xx:xx:76:xx:BE
    WLAN 5g (1) : 0A:xx:xx:76:xx:BE : 0A:xx:xx:76:xx:BF
    WLAN 5g (2) :                   : 0E:xx:xx:76:xx:BF
    WLAN 5g (3) :                   : 02:xx:xx:76:xx:BF

xavifr committed:
mediatek: filogic: add support ASUS RT-AX59U

(based on support for ASUS RT-AX59U by liushiyou006)

SOC: MediaTek MT7986
RAM: 512MB DDR4
FLASH: 128MB SPI-NAND (Winbond W25N01GV)
WIFI: Mediatek MT7986 DBDC 802.11ax 2.4/5 GHz
ETH: MediaTek MT7531 Switch
UART: 3V3 115200 8N1 (Pinout silkscreened / Do not connect VCC)

Upgrade from AsusWRT to OpenWRT using UART

    Download the OpenWrt initramfs image.
    Copy the image to a TFTP server reachable at 192.168.1.70/24. Rename the image to rtax59u.bin.

    Connect the PC with TFTP server to the RT-AX59U.
    Set a static ip on the ethernet interface of your PC.
    (ip address: 192.168.1.70, subnet mask:255.255.255.0)
    Conect to the serial console, interrupt the autoboot process by pressing '4' when prompted.

    Download & Boot the OpenWrt initramfs image.

    $ setenv ipaddr 192.168.1.1
    $ setenv serverip 192.168.1.70
    $ tftpboot 0x46000000 rtax59u.bin
    $ bootm 0x46000000

    Wait for OpenWrt to boot. Transfer the sysupgrade image to the device using scp and install using sysupgrade.

    $ sysupgrade -n <path-to-sysupgrade.bin>

Upgrade from AsusWRT to OpenWRT using WebUI

    Download transit TRX file from https://drive.google.com/drive/folders/1A20QdjK7Udagu31FSszpWAk8-cGlCwsq

    Upgrade firmware from WebUI (192.168.50.1) using downloaded TRX file

    Wait for OpenWRT to boot (192.168.1.1).

    Upgrade system with sysupgrade image using luci or uploading it through scp and executing sysupgrade command

MAC Address for WLAN 5g is not following the same algorithm as in AsusWRT.
We have increased by one the WLAN 5g to avoid collisions with other networks from WLAN 2g
when bit 28 is already set.

              : Stock             : OpenWrt
  WLAN 2g (1) : C8:xx:xx:0D:xx:D4 : C8:xx:xx:0D:xx:D4
  WLAN 2g (2) :                   : CA:xx:xx:0D:xx:D4
  WLAN 2g (3) :                   : CE:xx:xx:0D:xx:D4
  WLAN 5g (1) : CA:xx:xx:1D:xx:D4 : CA:xx:xx:1D:xx:D5
  WLAN 5g (2) :                   : CE:xx:xx:1D:xx:D5
  WLAN 5g (3) :                   : C2:xx:xx:1D:xx:D5

  WLAN 2g (1) : 08:xx:xx:76:xx:BE : 08:xx:xx:76:xx:BE
  WLAN 2g (2) :                   : 0A:xx:xx:76:xx:BE
  WLAN 2g (3) :                   : 0E:xx:xx:76:xx:BE
  WLAN 5g (1) : 0A:xx:xx:76:xx:BE : 0A:xx:xx:76:xx:BF
  WLAN 5g (2) :                   : 0E:xx:xx:76:xx:BF
  WLAN 5g (3) :                   : 02:xx:xx:76:xx:BF

Signed-off-by: Xavier Franquet <xavier@franquet.es>
(cherry picked from commit 782eb050082acac93c2f9b3eb22348234bc93e99)

xavifr commented:
network is not working yet

xavifr commented:
![image](https://github.com/openwrt/openwrt/assets/13116609/f9e3fdc4-0216-41b6-b9e3-db74071471b4)
![image](https://github.com/openwrt/openwrt/assets/13116609/835769ff-2cb2-4617-bf5a-1acceb144002)

csharper2005 commented:
First of all thanks for your work. I'm using 23.05.2 on this device for a few weeks and it works fine.

Did you copy the code from the main branch manually? https://github.com/openwrt/openwrt/pull/14326#issuecomment-1875956417

Backport PR title usually looks like that:
`[23.05] mediatek: add support for ASUS RT-AX59U`

xavifr commented:
> First of all thanks for your work. I'm using 23.05.2 on this device for a few weeks and it works fine.
> 
> Did you copy the code from the main branch manually? [#14326 (comment)](https://github.com/openwrt/openwrt/pull/14326#issuecomment-1875956417)
> 
> Backport PR title usually looks like that: `[23.05] mediatek: add support for ASUS RT-AX59U`

I cherry-picked, but forgot to use `-xs` params

csharper2005 commented:
@xavifr why the PR is still in draft? Do you need any help? 

And need to resolve merge conflict. Good opportunity to add missing cherry-pick info.

xavifr commented:
> @xavifr why the PR is still in draft? Do you need any help?
> 
> And need to resolve merge conflict. Good opportunity to add missing cherry-pick info.

Thanks @csharper2005, I forgot about draft state.

I've changed state to ready, rebased with openwrt-23.05 and added cherry-pick message.

G-Pereira commented:
> First of all thanks for your work. I'm using 23.05.2 on this device for a few weeks and it works fine.
> 
> Did you copy the code from the main branch manually? [#14326 (comment)](https://github.com/openwrt/openwrt/pull/14326#issuecomment-1875956417)
> 
> Backport PR title usually looks like that: `[23.05] mediatek: add support for ASUS RT-AX59U`

Good to know you're running 23.05.2 just fine. Can you send me the sysupgrade file?

csharper2005 commented:
@G-Pereira https://mega.nz/folder/xoBklZTZ#g1l_wH-nBOMONrDqOkeEvQ/folder/VlwWSTpY

G-Pereira commented:
@csharper2005 In that version and other builds I am testing I am having issues with WiFi on RT-AX59U. When I enable WiFi luci gets very slow and Wifi is not available:

![image](https://github.com/openwrt/openwrt/assets/15521303/f113e52a-9a01-4416-a889-2f92f62e49d9)

Any idea why?

csharper2005 commented:
@G-Pereira the main idea is to check kernel and system logs. And if necessary post them in device support topic https://forum.openwrt.org/t/openwrt-support-for-asus-rt-ax59u/154666
